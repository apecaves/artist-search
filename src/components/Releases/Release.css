.Release {
  display: flex;
  flex-direction: column;
  align-items: center;
  justify-content: center;
  text-align: center;
  color: black;
  text-decoration: none;
  &:focus,
  &:visited,
  &:link,
  &:active {
    text-decoration: none;
  }
  &:hover {
    color: grey;
  }
}
