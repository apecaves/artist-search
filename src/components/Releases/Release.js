import React from 'react';
import PropTypes from 'prop-types';

function Release({ release }) {
  const imgURL = `http://coverartarchive.org/release/${release.id}/front`;
  return (
    <>
      <p>{release.title}</p>
      <p>{release.date}</p>
      <p>{release.country}</p>
      <img src={imgURL} />
    </>
  );
}

Release.propTypes = {
  release: PropTypes.shape({
    title: PropTypes.string.isRequired,
    id: PropTypes.string.isRequired,
    date: PropTypes.string.isRequired,
    country: PropTypes.string.isRequired
  }).isRequired
};

export default Release;
