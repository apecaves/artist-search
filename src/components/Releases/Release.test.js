import React from 'react';
import { shallow } from 'enzyme';
import Release from './Release';

describe('Release component', () => {
  it('renders Release', () => {
    const wrapper = shallow(<Release release={{
      title: '', id: '', date: '', country: '', 'cover-art-archive': {}
    }} artistName={''}/>);
    expect(wrapper).toMatchSnapshot();
  });
});
  