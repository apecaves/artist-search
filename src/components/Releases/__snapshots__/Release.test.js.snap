// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Release component renders Release 1`] = `
<section
  className="Release"
>
  <Link
    className="Release"
    to="/release///"
  />
  <p />
  <p />
  <Link
    to="/release///"
  >
    <img
      src="/path/image"
    />
  </Link>
</section>
`;
